watchmode.go

```diff
@@ -0,0 +1,132 @@
+package templ
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+var developmentMode = os.Getenv("TEMPL_DEV_MODE") == "true"
+
+// WriteString writes the string to the writer. If development mode is enabled
+// s is replaced with the string at the index in the _templ.txt file.
+func WriteString(w io.Writer, index int, s string) (err error) {
+	if developmentMode {
+		_, path, _, _ := runtime.Caller(1)
+		if !strings.HasSuffix(path, "_templ.go") {
+			return errors.New("templ: attempt to use WriteString from a non templ file")
+		}
+		txtFilePath := strings.Replace(path, "_templ.go", "_templ.txt", 1)
+
+		literals, err := getWatchedStrings(txtFilePath)
+		if err != nil {
+			return fmt.Errorf("templ: failed to cache strings: %w", err)
+		}
+
+		if index > len(literals) {
+			return fmt.Errorf("templ: failed to find line %d in %s", index, txtFilePath)
+		}
+
+		s, err = strconv.Unquote(`"` + literals[index-1] + `"`)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = io.WriteString(w, s)
+	return err
+}
+
+// WriteWatchModeString is used when rendering templates in development mode.
+// the generator would have written non-go code to the _templ.txt file, which
+// is then read by this function and written to the output.
+// Deprecated: since templ v0.3.x generated code uses WriteString.
```

Comment:
Oddly, for linters to detect the `Deprecated` comment, it has to be at the start of a paragraph in the comment (https://go.dev/wiki/Deprecated). So this needs to either be:

```go
// WriteWatchModeString is used when rendering templates in development mode.
// the generator would have written non-go code to the _templ.txt file, which
// is then read by this function and written to the output.
//
// Deprecated: since templ v0.3.x generated code uses WriteString.
```

or

```go
// Deprecated: since templ v0.3.x generated code uses WriteString.
// WriteWatchModeString is used when rendering templates in development mode.
// the generator would have written non-go code to the _templ.txt file, which
// is then read by this function and written to the output.
```